src/lock_file/build_dispatch.rs

```diff
@@ -0,0 +1,387 @@
+use std::{path::Path, sync::Arc};
+
+use async_once_cell::OnceCell as AsyncCell;
+
+use once_cell::sync::OnceCell;
+
+use anyhow::Result;
+use pixi_manifest::EnvironmentName;
+use pixi_uv_conversions::{isolated_names_to_packages, names_to_build_isolation};
+use std::collections::HashMap;
+use tokio::runtime::Handle;
+use uv_build_frontend::SourceBuild;
+use uv_cache::Cache;
+use uv_client::RegistryClient;
+use uv_configuration::{
+    BuildKind, BuildOptions, BuildOutput, Concurrency, ConfigSettings, Constraints, IndexStrategy,
+    LowerBound, SourceStrategy,
+};
+use uv_dispatch::{BuildDispatch, SharedState};
+use uv_distribution_types::{
+    CachedDist, DependencyMetadata, IndexLocations, Resolution, SourceDist,
+};
+use uv_pep508::PackageName;
+use uv_pypi_types::Requirement;
+use uv_python::{Interpreter, PythonEnvironment};
+use uv_resolver::{ExcludeNewer, FlatIndex};
+use uv_types::{BuildContext, HashStrategy};
+
+use crate::{
+    activation::CurrentEnvVarBehavior,
+    project::{get_activated_environment_variables, Environment, EnvironmentVars},
+};
+
+use super::{conda_prefix_updater::CondaPrefixUpdated, CondaPrefixUpdater, PixiRecordsByName};
+
+/// This structure holds all the parameters needed to create a `BuildContext` uv implementation.
+pub struct UvBuildDispatchParams<'a> {
+    client: &'a RegistryClient,
+    cache: &'a Cache,
+    constraints: Constraints,
+    index_locations: &'a IndexLocations,
+    flat_index: &'a FlatIndex,
+    dependency_metadata: &'a DependencyMetadata,
+    shared_state: SharedState,
+    index_strategy: IndexStrategy,
+    config_settings: &'a ConfigSettings,
+    link_mode: uv_install_wheel::linker::LinkMode,
+    build_options: &'a BuildOptions,
+    hasher: &'a HashStrategy,
+    exclude_newer: Option<ExcludeNewer>,
+    bounds: LowerBound,
+    sources: SourceStrategy,
+    concurrency: Concurrency,
+}
+
+impl<'a> UvBuildDispatchParams<'a> {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        client: &'a RegistryClient,
+        cache: &'a Cache,
+        constraints: Constraints,
+        index_locations: &'a IndexLocations,
+        flat_index: &'a FlatIndex,
+        dependency_metadata: &'a DependencyMetadata,
+        shared_state: SharedState,
+        index_strategy: IndexStrategy,
+        config_settings: &'a ConfigSettings,
+        link_mode: uv_install_wheel::linker::LinkMode,
+        build_options: &'a BuildOptions,
+        hasher: &'a HashStrategy,
+        exclude_newer: Option<ExcludeNewer>,
+        bounds: LowerBound,
+        sources: SourceStrategy,
```

Comment:
Lets move the references together so we only need that on the `new` and then we can use `with_x` to add values for which defaults exist. That should make it a bit less arguments.

Reply:
Did this now and could remove a lot of unused params.